website/static/js/folderPickerNodeConfig.js

```diff
@@ -350,7 +350,10 @@ var FolderPickerViewModel = oop.defclass({
             self.nodeHasAuth(false);
             self.cancelSelection();
             self.currentDisplay(null);
+            $(self.folderPicker).html('');
+            self.loadedFolders(false);
             self.changeMessage(self.messages.deauthorizeSuccess(), 'text-warning', 3000);
+            $(self.folderPicker).folderpicker.prototype.flush();
```

Comment:
This is a bit awkward. We should be able to do `self.folderPicker.destroy()` here.